src/helpers/helpers.options.js

```diff
@@ -114,24 +114,32 @@ module.exports = {
 	 * @param {object} [context] - If defined and the current value is a function, the value
 	 * is called with `context` as first argument and the result becomes the new input.
 	 * @param {number} [index] - If defined and the current value is an array, the value
+	 * @param {object} [info] - object to return information about resolution in
```

Comment:
I think I'd initialize `info.cacheable` to `true` in this method so that if you call `resolve` outside of `_resolveDataElementOptions` then `info` contains a usable value. Then in `_resolveDataElementOptions` you can check `if (info.cacheable && !custom)` instead of `if (info.cacheable)`

Reply:
It would have to be something like
```js
if (info && info.cacheable === undefined) {
  info.cacheable = true;
}
```
because `resolve` is called in a loop. But I think its not needed. You'd need to provide the info object anyway, so not a biggie to provide initial value too.

Reply:
In other words, I like to keep the initialization (and any related if's) out of loop

Reply:
Maybe just document it then?
```
* @param {boolean} [info.cacheable] - Should be initialized to true. Will be updated by this method.
```